### PR TITLE
Added BackgorundChangerChallenge

### PR DESCRIPTION
**Pull request by Adidem23**
#  Added BackgroundColorChangerChallenge in list #296 
 

Title : Added BackgroundColorChangerChallenge in list

Issue No. : #296 

Code Stack :  React , Rapid Api , Axios , Javascript 

Close #<issue_no>

Video Tutorial for the challenge : 
https://www.loom.com/share/d4feadd3a9d7456e9887601102cdb7a1?sid=4537b4da-f843-4922-a724-603cb29057a5

# Checklist:

- [x] I have mentioned the issue number in my Pull Request.
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [x] I have updated the Index.html file for my contribution

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "frontend-mini-challenges",
+  "name": "fr",
```
Comment (sadanandpai) [inline package-lock.json:2]: why is this changed

**Review by sadanandpai — CHANGES_REQUESTED** [review]

File: react/src/machine-coding/BackgroundChanger/Background.jsx [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      method: 'GET',
+      url: 'https://colorful3.p.rapidapi.com/randomColor/2',
+      headers: {
+        'X-RapidAPI-Key': '4638530cfemshb5f9bea28be8a72p1d9bb8jsnef9a6f833608',
+        'X-RapidAPI-Host': 'colorful3.p.rapidapi.com'
+      }
+    };
+
+    try {
+      const response = await axios.request(options);
```
[the remark this answers is not in the document]
Comment (Adidem23) [reply react/src/machine-coding/BackgroundChanger/Background.jsx:18]: Okay i will use the fetch instead of the axios

File: package-lock.json (hunk 1 above)
Comment (Adidem23) [reply package-lock.json:2]: It was by my mistake I will fix this

File: react/src/helpers/challenges.ts [hunk 3]
```diff
@@ -51,6 +51,17 @@ export const challenges = new Map<string, Challenge>([
       tags: [],
     },
   ],
+  [
+    'BackgroundChanger',
+    {
+      title: 'BackgroundChanger',
+      link: 'BackgroundChanger',
```
[the remark this answers is not in the document]
Comment (Adidem23) [reply react/src/helpers/challenges.ts:58]: Okay I will change my folder name

File: react/src/helpers/challenges.ts [hunk 4]
```diff
@@ -51,6 +51,17 @@ export const challenges = new Map<string, Challenge>([
       tags: [],
     },
   ],
+  [
+    'BackgroundChanger',
+    {
+      title: 'BackgroundChanger',
```
[the remark this answers is not in the document]
Comment (Adidem23) [reply react/src/helpers/challenges.ts:57]: Okay No problem I will fix it !!

File: react/src/machine-coding/background-changer/Background.jsx [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    const options = {
+      method: 'GET',
+      headers: {
+        'X-RapidAPI-Key': '4638530cfemshb5f9bea28be8a72p1d9bb8jsnef9a6f833608',
+        'X-RapidAPI-Host': 'colorful3.p.rapidapi.com'
+      }
+    };
+
+    try {
+      const response = await fetch(url, options);
```
Comment (sadanandpai) [inline react/src/machine-coding/background-changer/Background.jsx:40]: for a new color to generate a normal function is enough. Do we need an API call to get a random color?

File: react/src/machine-coding/background-changer/background.module.css [hunk 6]
```diff
@@ -0,0 +1,99 @@
...
+    transition-property: border-radius, padding-top;
+    transition-duration: .2s;
+}
+
+.button:active .buttonbottom {
+    border-radius: 10px 10px 8px 8px / 8px;
+    padding-top: 0;
+}
+
+.buttonbase {
```
Comment (sadanandpai) [inline react/src/machine-coding/background-changer/background.module.css:87]: button-base or button_base or buttonBase

**Review by sadanandpai — CHANGES_REQUESTED** [review]

File: react/src/machine-coding/background-changer/Background.jsx (hunk 5 above)
Comment (Adidem23) [reply react/src/machine-coding/background-changer/Background.jsx:40]: I am using API for that so making API call . If you want I can simply Make an array and generate random color then but it will very bulky array so I am using api it generates random colour

File: react/src/machine-coding/background-changer/background.module.css (hunk 6 above)
Comment (Adidem23) [reply react/src/machine-coding/background-changer/background.module.css:87]: I have not understood What you want to say here  sorry Should I Change name of the class

Comment (sadanandpai) [reply react/src/machine-coding/background-changer/background.module.css:87]: yes

File: react/src/machine-coding/background-changer/Background.jsx (hunk 5 above)
Comment (sadanandpai) [reply react/src/machine-coding/background-changer/Background.jsx:40]: isn't it a simple function like below

```js
function getRandomColor() {
  var letters = '0123456789ABCDEF';
  var color = '#';
  for (var i = 0; i < 6; i++) {
    color += letters[Math.floor(Math.random() * 16)];
  }
  return color;
}
```

Comment (Adidem23) [reply react/src/machine-coding/background-changer/Background.jsx:40]: Ok I Will add it this in fetch fucntion then

File: react/src/machine-coding/background-changer/background.module.css (hunk 6 above)
Comment (Adidem23) [reply react/src/machine-coding/background-changer/background.module.css:87]: I will give buttonBase then for this className

Comment (Adidem23) [reply react/src/machine-coding/background-changer/background.module.css]: I Have Added Latest Changes you Mentioned @sadanandpai  sir

Comment (Adidem23) [reply react/src/machine-coding/background-changer/background.module.css]: Sorry For Making such mess for this simple task in repository !! . I am begineer so Sometimes I mess up really quick sorry once again sir @sadanandpai

**Review by sadanandpai — APPROVED** [review]
